config/main.py

```diff
@@ -4153,5 +4151,12 @@ def delete(ctx):
     sflow_tbl['global'].pop('agent_id')
     config_db.set_entry('SFLOW', 'global', sflow_tbl['global'])
 
+
+# Load plugins and register them
+helper = util_base.UtilHelper()
+for plugin in helper.load_plugins(plugins):
+    plugin.register(config)
```

Comment:
Given the plugins are not likely part of the "original" package, it would be nicer for the system to 
- Log the plugins being loaded
- Check with the plugin for compatibility (leaving to plugin how it decides) and log when the plugin reports incompatibility before registering.

Reply:
1. I've added a log message for plugin import and for plugin registration but with DEBUG log level so it will not appear by default in log otherwise it is going to be printed on every CLI command execution and may pollute the log.
2. Thanks for suggestion. Ideally, any compatibility checks should be done on installation time - here CLI auto generation based on YANG will help a lot (phase 2), thus, if package was able to be installed it is compatible.
There is another side of this question, if plugin happens to be incompatible and this is not reflected in version constraints in manifest the whole CLI is broken (possible because plugin will throw something like AttributeError or other kind of exception). I changed the code to catch Exception and log it as error, so the CLI will still be able to work but without new commands or changed commands. This does not catch all incompatibilities possible with this approach but better then nothing.
Given that, a plugin developer can raise Exception with message why he thinks it is not compatible which basically should satisfy your req no 2.